frontend/src/utilities/documents.ts

```diff
@@ -84,16 +93,49 @@ export default readonly(state);
 registerResponseHandler(ResponseType.UpdateOpenDocumentsList, (responseData: Response) => {
 	const documentListData = responseData as UpdateOpenDocumentsList;
 	if (documentListData) {
-		state.documents = documentListData.open_documents;
-		state.title = state.documents[state.activeDocumentIndex];
+		state.documents = documentListData.open_documents.map((name) => {
+			// Can't guarantee an exact match since it is going off of names
+			// This is ok though since it is only used to mark a document as unsaved
+			const probableMatch = state.documents.find((doc) => doc.name === name);
+			if (probableMatch) {
+				return { name, unsaved: probableMatch.unsaved, layerTreeHeight: probableMatch.layerTreeHeight };
```

Comment:
Let's change it so the Rust code sends all this information instead of just the names. We can pair program this part so I can show you the Rust stuff.